core/templates/services/questions-list.service.ts

```diff
@@ -87,6 +87,7 @@ export class QuestionsListService {
       this._questionSummariesForOneSkill = [];
       this._nextOffsetForQuestions = 0;
       this._moreQuestionsAvailable = true;
+      this._currentPage = 0;
```

Comment:
Can we call "resetPageNumber" before "getQuestionSummariesAsync" in cases where the current page is expected to be reset? It looks like this is how it is done in other places in the codebase.

Reply:
@kevintab95 Hey thanks for the review!, Looks like my initial understanding of why the bug was occurring was incorrect and the problematic call to `getQuestionSummariesAsync` was happening in `saveAndPublishQuestion`. I've added `resetPageNumber` there as requested. PTAL! 
https://github.com/oppia/oppia/pull/21044/commits/38a3292f9e001d075b2b11e875b0172a7ce3ec83